test/prompt/invocation_layer/test_hugging_face.py

```diff
@@ -0,0 +1,265 @@
+from unittest.mock import MagicMock, patch
+
+import pytest
+from transformers import (
+    AutoModelForSeq2SeqLM,
+    AutoTokenizer,
+    BloomForCausalLM,
+    T5ForConditionalGeneration,
+    T5TokenizerFast,
+    StoppingCriteriaList,
+    GenerationConfig,
+)
+
+from haystack.nodes.prompt.invocation_layer import HFLocalInvocationLayer
+from haystack.nodes.prompt.invocation_layer.handlers import HFTokenStreamingHandler, DefaultTokenStreamingHandler
+from haystack.nodes.prompt.invocation_layer.hugging_face import StopWordsCriteria
+
+
+@pytest.mark.unit
+def test_constructor_with_model_name():
+    """
+    Test that the constructor sets the pipeline with the model name (if provided)
+    """
+    layer = HFLocalInvocationLayer()
+
+    assert isinstance(layer.pipe.model, T5ForConditionalGeneration)
+    assert isinstance(layer.pipe.tokenizer, T5TokenizerFast)
+
+
+@pytest.mark.unit
+def test_constructor_with_model_name_and_device_map():
+    """
+    Test that the constructor sets the pipeline with the model name and device map
+    """
+
+    layer = HFLocalInvocationLayer(device_map="auto")
+
+    assert isinstance(layer.pipe.model, T5ForConditionalGeneration)
+    assert isinstance(layer.pipe.tokenizer, T5TokenizerFast)
```

Comment:
this test is only useful if you test the logic behind the assignment of `device`

Reply:
Testing that passing device_map and devices is ok. I'd love to improve these tests in general but for some reason, I can't patch/mock transformers methods like pipeline and get_task. All of the attempts to do so - fail. Perhaps it is something in transformers lib that needs to be done in a specific approach?

Reply:
> Perhaps it is something in transformers lib that needs to be done in a specific approach?

You tell me 😄 Please figure this out, we need to either mock that api successfully or understand why that's not possible and figure out an alternative.

Reply:
Silvano figured it out, and I took the idea from his unit test. Most of the tests are mocked except for stop words where I needed to hook into transformers to really check that stop words work as expected. Those tests are marked as integration.